### PR TITLE
feat(Coub): buttons, regex, external largeImageKey, elapsed timestamp

### PR DESCRIPTION
**Pull request by Venipa**
## Description 
- Fixed top page regex + presence data (null in buttons array wont display presence).
- Added largeImageKeys images (toggleable in settings) for coub channels & communities
- Added Elapsed timestamp (toggleable in settings)
- Removed youtube source buttons

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

New Settings (Show community/channel images & Show elapsed time):
image

Community Images:
image

Channel Images:
image

Playing Media:
image

</details>

### CODE REVIEW

File: websites/C/Coub/dist/metadata.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
 		"vi_VN": "Coub là trang web chia sẻ video trực tuyến. Nó cho phép người dùng tạo và chia sẻ video lặp lại lên tới mười giây, sử dụng các video có sẵn tại YouTube, Vimeo, hoặc tệp riêng của họ"
 	},
 	"url": "coub.com",
-	"version": "2.0.15",
+	"version": "2.0.16",
```
Comment (EncryptedDev) [inline websites/C/Coub/dist/metadata.json:22]: You have added functionality, this should be a minor bump instead
```suggestion
	"version": "2.1.0",
```

File: websites/C/Coub/dist/metadata.json [hunk 2]
```diff
@@ -40,6 +40,18 @@
 			"title": "Show current playing button",
 			"icon": "fas fa-external-link-alt",
 			"value": true
+		},
+		{
+			"id": "show_externalLargeImage",
+			"title": "Show community/channel images",
```
Comment (EncryptedDev) [inline websites/C/Coub/dist/metadata.json:46]: Is it possible to shorten this? Probably won't display too well

File: websites/C/Coub/dist/metadata.json [hunk 3]
```diff
@@ -40,6 +40,18 @@
...
+			"icon": "fas fa-image",
+			"value": true
+		},
+		{
+			"id": "show_startedBrowsing",
+			"title": "Show elapsed time",
+			"icon": "fas fa-stopwatch",
+			"value": true
 		}
```
Comment (EncryptedDev) [inline websites/C/Coub/dist/metadata.json:55]: What would be the reason why you wouldn't want this? Haven't seen many presences give this as a setting

File: websites/C/Coub/presence.ts [hunk 4]
```diff
@@ -322,26 +308,33 @@ presence.on("UpdateData", async () => {
 	const query: { [key: string]: unknown } = getQuery(),
 		context = pages.find(x => x.middleware(window, [query]));
 	if (!context) return;
+	const data: PresenceData = {
+			largeImageKey: "logo",
+		},
+		showStartedBrowsing = await presence
+			.getSetting<boolean>("show_startedBrowsing")
+			.catch(() => true);
```
Comment (EncryptedDev) [inline websites/C/Coub/presence.ts:316]: Why is the catch used here?

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/C/Coub/dist/metadata.json (hunk 2 above)
Comment (Venipa) [reply websites/C/Coub/dist/metadata.json:46]: thats why there is a tooltip?
image

File: websites/C/Coub/dist/metadata.json (hunk 3 above)
Comment (Venipa) [reply websites/C/Coub/dist/metadata.json:55]: someone requested to show elapsed time, i didnt liked it so i added a toggle

File: websites/C/Coub/presence.ts (hunk 4 above)
Comment (Venipa) [reply websites/C/Coub/presence.ts:316]: to return the default value,
getSetting can reject if its undefined
https://github.com/PreMiD/Extension/blob/5e9894de2c5e82d9425123ab353859158413e65d/src/util/presence/devHelper.ts#L411

File: websites/C/Coub/dist/metadata.json (hunk 2 above)
Comment (EncryptedDev) [reply websites/C/Coub/dist/metadata.json:46]: Changing the string to "Show Thumbnail" could be an easy solution here...

File: websites/C/Coub/dist/metadata.json (hunk 3 above)
Comment (EncryptedDev) [reply websites/C/Coub/dist/metadata.json:55]: Ok, use Title Case on the setting titles

File: websites/C/Coub/presence.ts (hunk 4 above)
Comment (EncryptedDev) [reply websites/C/Coub/presence.ts:316]: My apologies, but I see what you mean. Myself and Bas didn't realise this 😆

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**